File: exercism/javascript/trinary/trinary.js
const fp = require('lodash/fp')

class Trinary {
  constructor (repr) {
    const digits = repr.split('').reverse().map(fp.parseInt(10))
    const valid = !fp.some(fp.isNaN, digits)
    this.digits = valid ? digits : []
  }

  toDecimal () {
    const places = this.digits.map((d, i) => d * Math.pow(3, i))
    return fp.sum(places)
  }
}

module.exports = Trinary

File: exercism/csharp/trinary/Trinary.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class Trinary
{
    public static int ToDecimal (string digits)
    {
        if (digits.Except("012").Any()) return 0;
        return digits
          .Reverse()
          .Select((ch, i) => (int)Math.Pow(3, i) * (ch - '0'))
          .Sum();
    }
}
